### PR TITLE
[web][Store- Vue] Dynamic store registration

### PR DESCRIPTION
**Pull request by ann-aot**
# Issue Tracking

JIRA: https://aottech.atlassian.net/browse/FWF-728

# Changes

What are the main changes in PR?
Added dynamic store registration of the npm component in our package. now, there is no need for the client to register out the store in their main.ts

### CODE REVIEW

File: camunda-formio-tasklist-vue/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "camunda-formio-tasklist-vue",
-  "version": "1.0.7",
+  "version": "1.0.8",
```
Comment (kurianbenoy-aot) [inline camunda-formio-tasklist-vue/package.json:3]: need to update package-lock.json file as well

**Review by kurianbenoy-aot — COMMENTED** [review]
import serviceFlowModule from "../store/modules/serviceFlow-module";

Shouldn't the store details be changed in component `layout/LeftSide.vue` and `layout/Header.vue`?

https://github.com/AOT-Technologies/forms-flow-ai-extensions/blob/fix/store/camunda-formio-tasklist-vue/src/components/layout/LeftSider.vue#L113

File: camunda-formio-tasklist-vue/package.json [hunk 2]
```diff
@@ -77,4 +77,4 @@
     "*.json",
     "*.js"
   ]
-}
+}
```
Comment (kurianbenoy-aot) [inline camunda-formio-tasklist-vue/package.json:80]: Adding a blank line at end is recommended to remove the red color icons

File: camunda-formio-tasklist-vue/src/components/TaskList.vue [hunk 3]
```diff
@@ -347,20 +348,21 @@ const serviceFlowModule = namespace("serviceFlowModule");
 })
 export default class Tasklist extends Mixins(TaskListMixin) {
   @Prop() private getTaskId!: string;
+  @Prop() private mainStore!: any;
```
Comment (kurianbenoy-aot) [inline camunda-formio-tasklist-vue/src/components/TaskList.vue:351]: What is the default value for this mainStore prop? Can you update it `Views/Tasklist.vue` file as well

**Comment by kurianbenoy-aot** [comment]
Won't this change affect the existing docs on things to look out when integration? Can you make an update for that as well?

https://github.com/AOT-Technologies/forms-flow-ai-extensions/tree/master/camunda-formio-tasklist-vue#things-to-look-out-when-integrating

File: camunda-formio-tasklist-vue/src/components/TaskList.vue (hunk 3 above)
Comment (ann-aot) [reply camunda-formio-tasklist-vue/src/components/TaskList.vue:351]: thanks for bring this up, we no longer need this

**Comment by ann-aot** [comment]
> Won't this change affect the existing docs on things to look out when integration? Can you make an update for that as well?
> 
> https://github.com/AOT-Technologies/forms-flow-ai-extensions/tree/master/camunda-formio-tasklist-vue#things-to-look-out-when-integrating

yes, we need to update the doc

**Comment by ann-aot** [comment]
> Shouldn't the store details be changed in component `layout/LeftSide.vue` and `layout/Header.vue`?
> 
> https://github.com/AOT-Technologies/forms-flow-ai-extensions/blob/fix/store/camunda-formio-tasklist-vue/src/components/layout/LeftSider.vue#L113

I actually, didn't understand

File: camunda-formio-tasklist-vue/package-lock.json [hunk 4]
```diff
@@ -13384,4 +13384,4 @@
       }
     }
   }
-}
+}
```
Comment (kurianbenoy-aot) [inline camunda-formio-tasklist-vue/package-lock.json:13387]: Blank line at end

**Review by kurianbenoy-aot — APPROVED** [review]
Good job Ann 💯 . Everything looks good to me now 👍

**Review by krishnan-aot — APPROVED** [review]

**Pull request merged**